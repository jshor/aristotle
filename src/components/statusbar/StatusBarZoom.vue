<template>
  <input
    type="range"
    class="status-bar-zoom"
    :step="step"
    :min="min"
    :max="max"
    :value="value"
    @input="onInput"
  />
</template>

<script lang="ts">
import { defineComponent, computed } from 'vue'
import Icon from '@/components/Icon.vue'

export default defineComponent({
  name: 'StatusBarZoom',
  components: {
    Icon
  },
  props: {
    /** Current zoom value. */
    zoom: {
      type: Number,
      required: true
    },
    /** Minimum slider value. */
    min: {
      type: Number,
      default: 0
    },
    /** Maximum slider value. */
    max: {
      type: Number,
      default: 200
    },
    /** Slider increment value. */
    step: {
      type: Number,
      default: 10
    }
  },
  setup (props, { emit }) {
    const value = computed(() => Math.round(props.zoom * 100))

    /**
     * Slider input event handler.
     *
     * @emits zoom as a decimal percentage factor [0.0, 1.0]
     */
    function onInput ($event: Event) {
      const target = $event.target as HTMLInputElement
      const zoom = parseFloat(target.value)

      emit('zoom', zoom / 100)
    }

    return {
      onInput,
      value
    }
  }
})
</script>

<style lang="scss">
.status-bar-zoom {
  -webkit-appearance: none;
  height: 0.5em;
  background-color: var(--color-bg-quaternary);
  transition: opacity .2s;
  opacity: 1;

  &:hover {
    opacity: 0.7;
  }

  &::-webkit-slider-thumb {
    -webkit-appearance: none;
    appearance: none;
    width: 0.5em;
    height: 1.5em;
    background-color: var(--color-secondary);
    cursor: ew-resize;
  }

  &::-moz-range-thumb {
    width: 0.5em;
    height: 1.5em;
    background-color: var(--color-secondary);
    cursor: ew-resize;
  }
}
</style>
