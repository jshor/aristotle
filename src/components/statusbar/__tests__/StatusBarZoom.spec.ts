import { shallowMount, VueWrapper } from '@vue/test-utils'
import { ComponentPublicInstance } from 'vue'
import StatusBarZoom from '../StatusBarZoom.vue'

describe('Oscilloscope Title Bar', () => {
  let wrapper: VueWrapper

  beforeEach(() => {
    wrapper = shallowMount(StatusBarZoom, {
      props: {
        zoom: 1
      }
    })
  })

  it('should match the snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should emit `zoom` as a decimal zoom factor', async () => {
    await wrapper.setProps({ clearable: true })
    const slider = wrapper.findAll('input')[0]

    slider.element.value = '110'
    slider.trigger('input')

    expect(wrapper.emitted()).toHaveProperty('zoom')
    expect(wrapper.emitted().zoom[0]).toEqual([1.1])
  })
})
