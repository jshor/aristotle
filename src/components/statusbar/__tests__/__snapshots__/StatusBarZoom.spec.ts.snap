// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Oscilloscope Title Bar should match the snapshot 1`] = `"<input type="range" class="status-bar-zoom" step="10" min="0" max="200">"`;
